packages/analytics-core/src/plugins/destination.ts

```diff
@@ -59,7 +59,6 @@ export class Destination implements DestinationPlugin {
 
     this.storageKey = `${STORAGE_PREFIX}_${this.config.apiKey.substring(0, 10)}`;
     const unsent = await this.config.storageProvider?.get(this.storageKey);
-    this.saveEvents(); // sets storage to '[]'
```

Comment:
In the Destination plugin, we originally clear the storage and copy all events into a local variable queue(line 62). And when we get a response, we put them back into the queue for retry and save the events back to the storage(line 105).